src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CreateExportRequestHandler.cs

```diff
@@ -59,7 +61,16 @@ public async Task<CreateExportResponse> Handle(CreateExportRequest request, Canc
 
             if (outcome == null)
             {
-                var jobRecord = new ExportJobRecord(request.RequestUri, hash, requestorClaims);
+                // Remove the connection settings from the request URI store it in the secret store.
+                NameValueCollection queryParameters = HttpUtility.ParseQueryString(request.RequestUri.Query);
+
+                queryParameters.Remove(KnownQueryParameterNames.DestinationType);
```

Comment:
Because the connection settings contains secret, we should remove it from the RequestURI so that it's not stored in the datastore in plain text.

The [spec](https://github.com/HL7/bulk-data/blob/master/spec/export/index.md#response---complete-status) requires the original URI to be returned when the job is completed. However, we delete the secret after the job is completed so we won't able to add these parameters back. We could choose to store the secret indefinitely but that might be a security concern. I think it's a fine compromise to return the original URL without the secret. What do you think @hansenms?

Reply:
For now, I am going to implement the way it is. If we have feedback that the original URL must be reconstructed, we can fix it at that point.